darts/preprocessing/base_transformer.py

```diff
@@ -0,0 +1,239 @@
+
+        Parameters
+        ----------
+        data
+            Object which will be transformed.
+        args
+            Positional arguments for the `transform` method
+        kwargs
+            Keyword arguments for the `transform` method
+
+        Returns
+        -------
+        T
+            Transformed data.
+        """
+        raise NotImplementedError(f"transform not implemented for transformer {self.name}")
+
+    def inverse_transform(self, data: T, *args, **kwargs) -> T:
+        """
+        Inverse transformation of data. Not implemented in base class.
+
+        Parameters
+        ----------
+        data
+            Object which will be inverse transformed.
+        args
+            Positional arguments for the `inverse_transform` method
+        kwargs
+            Keyword arguments for the `inverse_transform` method
+
+        Returns
+        -------
+        T
+            Inverse transformed data.
+        """
+        raise NotImplementedError(f"inverse_transform not implemented for transformer {self.name}")
+
+    def predict(self, data: T, *args, **kwargs) -> T:
+        """
+        Predict function. Not implemented in base class.
+
+        Parameters
+        ----------
+        data
+            Object on which predict will be performed.
+        args
+            Positional arguments for the `predict` method
+        kwargs
+            Keyword arguments for the `predict` method
+
+        Returns
+        -------
+        T
+            Predicted values.
+        """
+        raise NotImplementedError(f"predict not implemented for transformer {self.name}")
+
+    def fit(self, data: T) -> 'BaseTransformer[T]':
+        """
+        Function which will fit transformer. Not implemented in base class.
+
+        Parameters
+        ----------
+        data
+            Object on which transformer will be fitter.
+
+        Returns
+        -------
+        BaseTransformer[T]
+            Fitted transformer.
+        """
+        raise NotImplementedError(f"fit not implemented for transformer {self.name}")
+
+    def fit_transform(self, data: T, *args, **kwargs) -> T:
+        """
+        First fit transformer with data and then perform transformation on data.
+
+        Parameters
+        ----------
+        data
+            Object used to fit and transform.
+        args
+            Positional arguments for the `transform` method
+        kwargs
+            Keyword arguments for the `transform` method
+
+        Returns
+        -------
+        T
+            Transformed data.
+        """
+        return self.fit(data).transform(data, *args, **kwargs)
+
+    def __call__(self, data: T, *args, fit: bool = False, inverse: bool = False, **kwargs) -> T:
+        """
+        Calling transformer will run validation and perform transformation or
+        inverse_transformation on data. If fit flag is set transformer will first be fitted.
```

Comment:
Is this really needed ?

I find it less intuitive/clear to write `transformer(data, inverse=true)` than `transformer.inverse_transform(data)`